pulsar-jms/src/main/java/com/datastax/oss/pulsar/jms/PulsarJMSProducer.java

```diff
@@ -191,10 +191,18 @@ private void getProducerAndSend(Destination destination, Message message) throws
     for (Map.Entry<String, Object> prop : properties.entrySet()) {
       message.setObjectProperty(prop.getKey(), prop.getValue());
     }
-    message.setJMSPriority(priority);
-    message.setJMSCorrelationIDAsBytes(correlationID);
-    message.setJMSType(jmsType);
-    message.setJMSReplyTo(jmsReplyTo);
+    if (message.getJMSPriority() != Message.DEFAULT_PRIORITY) {
+      message.setJMSPriority(priority);
+    }
+    if (message.getJMSCorrelationIDAsBytes() != null) {
+      message.setJMSCorrelationIDAsBytes(message.getJMSCorrelationIDAsBytes());
+    }
+    if (message.getJMSType() != null) {
+      message.setJMSType(message.getJMSType());
+    }
+    if (message.getJMSReplyTo() != null) {
+      message.setJMSReplyTo(message.getJMSReplyTo());
+    }
```

Comment:
```suggestion
    if (message.getJMSCorrelationIDAsBytes() == null || correlationID != null) {
      message.setJMSCorrelationIDAsBytes(correlationID);
    }
    if (message.getJMSType() == null || jmsType != null) {
      message.setJMSType(jmsType);
    }
    if (message.getJMSReplyTo() == null || jmsReplyTo != null) {
      message.setJMSReplyTo(jmsReplyTo);
    }
```

I don't know if this is a correct fix, but otherwise this change makes no sense - it gets value from a message and sets it back.
Proposed change avoids overwrite of a value if it is set and producer does not have corresponding value.